src/battle-animations.ts

```diff
@@ -2554,117 +2554,120 @@ class PokemonSprite extends Sprite {
+			rage: ['Rage', 'neutral'],
+			roost: ['Landed', 'neutral'],
+			protect: ['Protect', 'good'],
+			quickguard: ['Quick Guard', 'good'],
+			wideguard: ['Wide Guard', 'good'],
+			craftyshield: ['Crafty Shield', 'good'],
+			matblock: ['Mat Block', 'good'],
+			maxguard: ['Max Guard', 'good'],
+			helpinghand: ['Helping Hand', 'good'],
+			magiccoat: ['Magic Coat', 'good'],
+			destinybond: ['Destiny Bond', 'good'],
+			snatch: ['Snatch', 'good'],
+			grudge: ['Grudge', 'good'],
+			charge: ['Charge', 'good'],
+			endure: ['Endure', 'good'],
+			focuspunch: ['Focusing', 'neutral'],
+			shelltrap: ['Trap set', 'neutral'],
+			powder: ['Powder', 'bad'],
+			electrify: ['Electrify', 'bad'],
+			ragepowder: ['Rage Powder', 'good'],
+			followme: ['Follow Me', 'good'],
+			instruct: ['Instruct', 'neutral'],
+			beakblast: ['Beak Blast', 'neutral'],
+			laserfocus: ['Laser Focus', 'good'],
+			spotlight: ['Spotlight', 'neutral'],
+			itemremoved: null,
 			// partial trapping
-			bind: '<span class="bad">Bind</span>',
-			clamp: '<span class="bad">Clamp</span>',
-			firespin: '<span class="bad">Fire Spin</span>',
-			infestation: '<span class="bad">Infestation</span>',
-			magmastorm: '<span class="bad">Magma Storm</span>',
-			sandtomb: '<span class="bad">Sand Tomb</span>',
-			snaptrap: '<span class="bad">Snap Trap</span>',
-			thundercage: '<span class="bad">Thunder Cage</span>',
-			whirlpool: '<span class="bad">Whirlpool</span>',
-			wrap: '<span class="bad">Wrap</span>',
+			bind: ['Bind', 'bad'],
+			clamp: ['Clamp', 'bad'],
+			firespin: ['Fire Spin', 'bad'],
+			infestation: ['Infestation', 'bad'],
+			magmastorm: ['Magma Storm', 'bad'],
+			sandtomb: ['Sand Tomb', 'bad'],
+			snaptrap: ['Snap Trap', 'bad'],
+			thundercage: ['Thunder Cage', 'bad'],
+			whirlpool: ['Whirlpool', 'bad'],
+			wrap: ['Wrap', 'bad'],
 			// Gen 1
-			lightscreen: '<span class="good">Light&nbsp;Screen</span>',
-			reflect: '<span class="good">Reflect</span>',
+			lightscreen: ['Light Screen', 'good'],
+			reflect: ['Reflect', 'good'],
 		};
 		for (let i in pokemon.volatiles) {
-			if (typeof statusTable[i] === 'undefined') status += '<span class="neutral">[[' + i + ']]</span>';
-			else status += statusTable[i];
+			status += statusTable[i] !== null ? PokemonSprite.buildEffectTag(statusTable[i] || i) : '';
 		}
 		for (let i in pokemon.turnstatuses) {
 			if (i === 'roost' && !pokemon.getTypeList().includes('Flying')) continue;
-			if (typeof statusTable[i] === 'undefined') status += '<span class="neutral">[[' + i + ']]</span>';
-			else status += statusTable[i];
+			status += statusTable[i] !== null ? PokemonSprite.buildEffectTag(statusTable[i] || i) : '';
 		}
 		for (let i in pokemon.movestatuses) {
-			if (typeof statusTable[i] === 'undefined') status += '<span class="neutral">[[' + i + ']]</span>';
-			else status += statusTable[i];
+			status += statusTable[i] !== null ? PokemonSprite.buildEffectTag(statusTable[i] || i) : '';
```

Comment:
It seems like you could move more of this logic into `buildEffectTag`.

Like

```suggestion
			status += PokemonSprite.buildEffectTag(i);
```

Reply:
Where would I put the `statusTable` declaration? I wouldn’t want to redeclare it every time the method is called, so should it be a property on `PokemonSprite`?

Reply:
actually, what if I make it a static property of the `PokmeonSprite` prototype, and after each status is built I store the result in the prototype property to cache it? seems like a good blend of readability and performance.

Reply:
Yeah, it can be a static property. Storing the result in the prototype property to cache it, though, is I think an unnecessary hyper-optimization not worth the loss in readability.

(But if you really want to do it, I won't stop you.)